src/dali/plugin/input/rest/kraken.py

```diff
@@ -0,0 +1,451 @@
+from dali.in_transaction import InTransaction
+from dali.intra_transaction import IntraTransaction
+from dali.out_transaction import OutTransaction
+from dali.configuration import Keyword, _FIAT_SET
+from dali.cache import load_from_cache, save_to_cache
+
+# keywords
+_AMOUNT: str = "amount"
+_ASSET: str = "asset"
+_BASE: str = "base"
+_BASE_ID: str = "baseId"
+_COST: str = "cost"
+_COUNT: str = "count"
+_CREDIT: str = "credit"
+_DEPOSIT: str = "deposit"
+_FEE: str = "fee"
+_ID: str = "id"
+_IN: str = "in"
+_INTRA: str = "intra"
+_LEDGER: str = "ledger"
+_MARGIN: str = "margin"
+_OFFSET: str = "ofs"
+_OUT: str = "out"
+_PAIR: str = "pair"
+_PRICE: str = "price"
+_QUOTE: str = "quote"
+_REFID: str = "refid"
+_RESULT: str = "result"
+_ROLLOVER: str = "rollover"
+_SALE: str = "sale"
+_SETTLED: str = "settled"
+_STAKING: str = "staking"
+_TIMESTAMP: str = "time"
+_TRADE: str = "trade"
+_TRADES: str = "trades"
+_TRANSFER: str = "transfer"
+_TYPE: str = "type"
+_WITHDRAWAL: str = "withdrawal"
+
+# Record Limits
+_TRADE_RECORD_LIMIT: int = 50
+
+_KRAKEN_FIAT_SET: Set[str] = {"AUD", "CAD", "EUR", "GBP", "JPY", "USD",
+                              "USDC", "USDT", "ZAUD", "ZCAD", "ZEUR", "ZGBP", "ZJPY", "ZUSD"}
+
+_KRAKEN_FIAT_LIST = list(set(list(_KRAKEN_FIAT_SET) + list(_FIAT_SET)))
+
+
+class InputPlugin(AbstractCcxtInputPlugin):
+    __EXCHANGE_NAME: str = "kraken"
+    __PLUGIN_NAME: str = "kraken_REST"
+    __DEFAULT_THREAD_COUNT: int = 1
+    __CACHE_FILE: str = "kraken.pickle"
+
+    def __init__(
+        self,
+        account_holder: str,
+        api_key: str,
+        api_secret: str,
+        native_fiat: str,
+        thread_count: Optional[int] = __DEFAULT_THREAD_COUNT,
+        use_cache: Optional[bool] = True,
+    ) -> None:
+        self.__api_key = api_key
+        self.__api_secret = api_secret
+
+        # We will have a default start time of July 27th, 2011 since Kraken Exchange officially launched on July 28th.
+        super().__init__(account_holder, datetime(2011, 7, 27, 0, 0, 0, 0), native_fiat, thread_count)
+        self.__logger: logging.Logger = create_logger(f"{self.__EXCHANGE_NAME}")
+        self.base_id_to_base: Dict[str, str] = {}
+        self.use_cache: Optional[bool] = use_cache
+        self._initialize_client()
+
+    def exchange_name(self) -> str:
+        return self.__EXCHANGE_NAME
+
+    def plugin_name(self) -> str:
+        return self.__PLUGIN_NAME
+
+    def _initialize_client(self) -> kraken:
+        return kraken(
+            {
+                "apiKey": self.__api_key,
+                "enableRateLimit": True,
+                "secret": self.__api_secret,
+            }
+        )
+
+    def _initilaize_markets(self) -> None:
+        self._client.load_markets()
+        self._client.markets_by_id.update({"BSVUSD": {_ID: "BSVUSD", _BASE_ID: "BSV", _BASE: "BSV", _QUOTE: "USD"}})
+
+        for dummy_key, value in self._client.markets_by_id.items():
```

Comment:
Maybe you can be more descriptive and just use `values()`:
```
for market in self._client.markets_by_id.values():
```

Reply:
that's probably fine

Reply:
Thanks for contributing to this project. We are really thankful, but keep in mind that this code could possibly be used for years and that other developers will most likely edit and update it. Keeping it clean and readable will help them.

If it is too much for you though, I can clean it up once you've submitted the code.

Reply:
I understand and agree to the change.

Reply:
I think you guys don't like the use of the word dummy. :)

I do it because mypy understands to ignore unused variables with that naming convention.

Reply:
It just reminds me of a quote.

“Perfection is achieved, not when there is nothing more to add, but when there is nothing left to take away.”
― Antoine de Saint-Exupéry

This project is small at the moment, but it could grow in size with several developers that are only spending a few hours a week on it (like me). You want the code to be easily approachable and void of clutter so that they can jump in and understand exactly what is going on.

Also, you are basically admitting to using a workaround for a safeguard (mypy) put in place to preserve data integrity. :)  

I mean my work desk is a total mess, but for some reason, I like code that is concise and compact, yet easy to understand. I think most coders feel the same way.

Reply:
I appreciate your input and I don't want to get bogged down in a debate. I personally like clean and concise code, but I also understand that different people have different opinions.  I don't think it's fair to assume that all coders have the same preferences.  I think it's important to prioritize getting the code working before focusing on cleaning it up.  I have limited time as well and I spent a lot of time on this and I'm frustrated that it took so long. Part of the reason was the hours the pair converter would take to resolve transactions which would affect iteration time to resolve bugs (including the spot_price bug).  I have been open to making changes and I think I have shown this through the code review process. The code review process is an important part of becoming a better coder. When you mentioned the suggestion I thought it was good, that's why I accepted it. Additionally, I wasn't "basically" admitting - I was admitting. Oh and I meant to say pylint (not mypy - whoops!).